README.md

```diff
@@ -1,48 +1,105 @@
+foundations of the [Snowflake Native App Framework][Native Apps docs]. 
 
-These templates do not restrict or limit developers. Instead, the templates provide examples of how to use core Snowflake features to ingest data and encapsulate application code within a Snowflake Native App. 
+**In order to accelerate and standardize the process of developing new Native Connectors, Snowflake would like to introduce
+the Native SDK for Connectors which is a set of libraries, developing tools, example connectors and connector template 
+projects.**
 
-The Snowflake Native App Framework allows providers to publish and monetize a Snowflake Native App on the Snowflake marketplace. Snowflake Native App developers can clone the template repository, modify the boilerplate code, and create their own Snowflake Connectors.
+## Key components
 
-## Pull-based connector templates and quickstarts
+In the diagram below, you can find key components that are listed with a brief description below the diagram.
 
-### When to use a pull-based pattern
+![Repository Components Diagram](./.assets/components_diagram.png)
 
-Pull-based patterns are effective when the source data provider does not manage customer data in Snowflake and is not willing to incur COGS for a continuous data share in Snowflake. These patterns are also effective when a source data provider has well-documented APIs that customers can use to replicate and consume data.
+### Native SDK for Connectors Java
 
-### How to use a pull-based pattern
+The Java library distributed through the [Maven Central Repository][SDK in Maven Central] which consists of .sql files 
+that deliver database objects definitions that are created during the installation of the Connector in the Snowflake 
+environment and a bunch of Java classes that mainly serve as procedure handlers. However, among these Java classes, 
+there also are some helper/util classes that help in coping with most of the common problems that might be encountered 
+during the development of the Connector project. The database objects and Java classes create a coherent whole that 
+makes managing the state of the Connector much easier and lets the developers focus on the implementation of the 
+specific external source ingestion logic which also is easier to do with the SDK.
 
-By using a pull-based connector pattern, providers (Snowflake, or a third-party ETL provider) can publish, distribute a Native Connector based on a Snowflake Native App using the Snowflake Marketplace. A Native Connector uses direct external access to connect with the source application. A Native Connector performs outbound authentication, fetches data from the source directly into a customer account, processes and persists the data based on user specified configuration.
+Learn more from the [official documentation][Native SDK official docs].
 
-![image Pull-Based Connector Architecture](./images/pull_based_connector_architecture.png)
+#### Task Reactor
```

Comment:
actually why do we mention task reactor here if it's included in the jar? Using this logic we could describe every component included in SDK